examples/spec/integration/register_namespace_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'workflows/timeout_workflow'
+
+describe 'Temporal.register_namespace' do
+  it 'can register a new namespace' do
+    # have to generate a new namespace on each run because currently can't delete namespaces
+    name = "test_namespace_#{SecureRandom.uuid}"
+    description = 'this is the description'
+    retention_period = 30
+    namespace_data = { test: 'value' }
+
+    result = Temporal.register_namespace(name, description, retention_period: retention_period, namespace_data: namespace_data)
+    expect(result).to be_an_instance_of(Temporal::Api::WorkflowService::V1::RegisterNamespaceResponse)
+
+    # fetch the namespace from Temporal and check it exists and has the correct settings
+    found_namespace = nil
+    next_page_token = ''
+
+    while found_namespace.nil?
+      result = Temporal.list_namespaces(page_size: 100, next_page_token: next_page_token)
+      result.namespaces.each do |namespace|
+        if namespace.namespace_info.name == name
+          found_namespace = namespace
+          break
+        end
+      end
+
+      if result.next_page_token == ''
+        break
+      else
+        next_page_token = result.next_page_token
+      end
+    end
+
+    expect(found_namespace).to_not eq(nil)
+    expect(found_namespace.namespace_info.name).to eq(name)
+    expect(found_namespace.namespace_info.data).to eq(namespace_data)
+    expect(found_namespace.config.workflow_execution_retention_ttl.seconds).to eq(retention_period * 24 * 60 * 60)
```

Comment:
If behind the scenes the retention TTL is converted to seconds, why not try and pass a small float value so that retention seconds is under 5; then you could test to see if the namespace disappears after the retention seconds expire.

I know the method signature says it takes "integer days" as the arg, but there's no reason you can't break that for testing purposes. Plus, there's no code here to a) confirm the arg is an integer, or b) convert it to an integer. That said, 0 is also an integer so that would make the namespace expire immediately, yes?

Reply:
Yeah it's not the namespace TTL but rather how long Temporal will keep workflow histories after they've finished running. I could add this test but it feels like it's testing an implementation detail of Temporal (we're testing an internal mechanic of the server). I know the case could be made that is what integration tests are but I invisigned more along the lines:
1. Check that the server can actually accept the request
2. Check the server responds with sensible values
Rather than testing exactly what the server implementation does with the config! Open to discussion here tho

Reply:
Ha, just learned something. Okay, I guess this part doesn't matter.